haproxy/Dockerfile

```diff
@@ -0,0 +1,78 @@
+FROM ghcr.io/cybozu/ubuntu:20.04 AS build
+
+ENV HAPROXY_BRANCH 2.6
+ENV HAPROXY_MINOR 2.6.10
+ENV HAPROXY_SHA256 e71b2cd9ca1043345f083a5225078ccf824dced2b5779d86f11fa4e88f451773
+ENV HAPROXY_SRC_URL http://www.haproxy.org/download
+ENV DEBIAN_FRONTEND noninteractive
+
+# if you use ubuntu:22.04, you must use "libssl3=3.0.2-0ubuntu1.9". but HA-Proxy 2.6 does NOT adapt it.
+# We choose "ubuntu:20.04" and "libssl1.1" to get a stable operation.
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    procps \
+    zlib1g \
+    libssl1.1 \
+    "libpcre2-*" \
+    liblua5.3-0 \
+    libatomic1 \
+    tar \
+    curl \
+    socat \
+    gcc \
+    make \
+    libc6-dev \
+    libssl-dev \
+    libpcre3-dev \
+    zlib1g-dev \
+    liblua5.3-dev && \
+    curl -sfSL "${HAPROXY_SRC_URL}/${HAPROXY_BRANCH}/src/haproxy-${HAPROXY_MINOR}.tar.gz" -o haproxy.tar.gz && \
+    echo "$HAPROXY_SHA256 *haproxy.tar.gz" | sha256sum -c - && \
+    mkdir -p /tmp/haproxy && \
+    tar -xzf haproxy.tar.gz -C /tmp/haproxy --strip-components=1 && \
+    rm -f haproxy.tar.gz && \
+    make -C /tmp/haproxy -j"$(nproc)" TARGET=linux-glibc CPU=generic USE_PCRE2=1 USE_PCRE2_JIT=1 USE_OPENSSL=1 \
+                            USE_TFO=1 USE_LINUX_TPROXY=1 USE_LUA=1 USE_GETADDRINFO=1 \
+                            USE_PROMEX=1 USE_SLZ=1 \
+                            all && \
+    make -C /tmp/haproxy TARGET=linux-glibc install-bin install-man
+
+
+FROM ghcr.io/cybozu/ubuntu:20.04
+
+LABEL org.opencontainers.image.source="https://github.com/cybozu/neco-containers/haproxy"
+
+ENV HAPROXY_UID haproxy
+ENV HAPROXY_GID haproxy
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN mkdir -p /var/lib/haproxy && \
+    mkdir -p /usr/local/etc/haproxy
+
+COPY --from=build /usr/local/sbin/haproxy /usr/sbin/haproxy
```

Comment:
neco-containers often handles the following.

1. Place binaries built from source code under `/usr/loca/<package>`.
2. Copy the license file under `/usr/local/<package>/LICENSE`

```
$ docker run -it quay.io/cybozu/argocd:2.6
argocd@dd08a2f22ea7:~$ which argocd
/usr/local/bin/argocd
argocd@dd08a2f22ea7:~$ ls /usr/local/argocd/
LICENSE
```

Reply:
I follow it.